design/kubebuilder-migration.md

```diff
@@ -0,0 +1,64 @@
+# Move Contour to use kubebuilder instead of informers
+
+Status: Draft
+
+Contour's method of interaction with Kubernetes is two years old, and the community has moved on from Informers.
+This proposal suggests that we should move to using Kubebuilder [controller-runtime][1] for interacting with Kubernetes.
+This will make extending Contour with additional CRDs as we need to more straightforward.
+
+## Goals
+
+- Migrate Contour to use Kubebuilder [controller-runtime][1] instead of informers.
```

Comment:
Are you proposing to use kubebuilder or just the controller-runtime project?

Reply:
Just the controller-runtime parts, since we don't need the autogeneration scaffolding.

Reply:
It might be good to clarify that, perhaps in the non goals section.